packages/tpetra/core/src/Tpetra_Import_Util2.hpp

```diff
@@ -183,9 +199,11 @@ sortCrsEntries (const Teuchos::ArrayView<size_t> &CRS_rowptr,
         for(Ordinal k = j; k >= 0; k-=m) {
           if(locIndices[k+m] >= locIndices[k])
             break;
-          Scalar dtemp = locValues[k+m];
-          locValues[k+m] = locValues[k];
-          locValues[k] = dtemp;
+          if (!Teuchos::is_null(locValues)) {
```

Comment:
It might be easier for the compiler to optimize this if you would lift that test out of the inner loop.  For example:
```
const bool permute_values_array = ! Teuchos::is_null (locValues);
while (m > 0) {
  // ...
  if (permute_values_array) {
    Scalar dtemp = locValues[k+m];
    // ...
  }
  // ...
}
```